### PR TITLE
Add fix for BottomNavigationBarItem.title -> label

### PR DESCRIPTION
**Pull request by Piinks**
This adds a fix for the deprecated BottomNavigationBarItem.title to migrate to label
Deprecated in #59127

Part of #74908

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/test_fixes/cupertino.dart [hunk 1]
```diff
@@ -123,4 +123,8 @@ void main() {
   CupertinoTextField.borderless(maxLengthEnforced: false);
   final CupertinoTextField textField;
   textField.maxLengthEnforced;
+
+  // Changes made in https://github.com/flutter/flutter/pull/59127
+  const BottomNavigationBarItem bottomNavigationBarItem = BottomNavigationBarItem(title: myTitle);
+  newTitle = bottomNavigationBarItem.title;
```
Comment (goderbauer) [inline packages/flutter/test_fixes/cupertino.dart:129]: nit: this isn't valid dart since "newTitle" is undeclared?

Comment (Piinks) [reply packages/flutter/test_fixes/cupertino.dart:129]: mmmmmmmmm that is kind of true. `label` (`String?`) and `title` (`Widget?`) are not of the same type. 😒 
I was wondering if this is another fix we might *not* want to provide. I was curious what your thoughts were. I was about to comment about it. You do not miss a beat. lol

Comment (Piinks) [reply packages/flutter/test_fixes/cupertino.dart:129]: Definitely a 'partial' fix.

Comment (goderbauer) [reply packages/flutter/test_fixes/cupertino.dart:129]: I think we decided with product management that we want these partial fixes, right?

To make it valid dart code in the test, you could declare `newTitle` as `Object?` or leave the left side out altogether like we do in line 125 above?

Comment (Piinks) [reply packages/flutter/test_fixes/cupertino.dart:129]: Yeah, I figured. Didn't know if it was worth evaluating on a case by case basis. :)
I'll update.

**Review by goderbauer — APPROVED** [review]
LGTM

**Pull request merged**